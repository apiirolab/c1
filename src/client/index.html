<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>