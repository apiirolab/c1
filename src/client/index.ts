export const info = 'Client Info';
