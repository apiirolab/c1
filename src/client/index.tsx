import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
