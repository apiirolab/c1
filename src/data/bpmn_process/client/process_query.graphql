query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
