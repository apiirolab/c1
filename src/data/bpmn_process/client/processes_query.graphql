query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
