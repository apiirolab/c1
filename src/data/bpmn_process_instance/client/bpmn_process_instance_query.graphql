fragment BpmnProcessResource on Resource {
  name
  type
  id
  content
}

fragment BpmnProcessComment on Comment {
  id
  date
  replyTo
  text
  user {
    name
  }
}

fragment BpmnProcessTask on BpmnTaskInstance {
  name
  dateStarted
  dateFinished
  performer {
    name
  }
  performerRoles
}

fragment BpmnProcessDefinition on BpmnProcess {
  name
  description
  type
  model
  resources {
    ...BpmnProcessResource
  }
}

query BpmnProcessInstanceQuery($id: String!) {
  bpmnProcessInstance(id: $id) {
    id
    status
    process {
      ...BpmnProcessDefinition
    }
    owner {
      name
    }
    tasks {
      ...BpmnProcessTask
    }
    comments {
      ...BpmnProcessComment
    }
    resources
    dateStarted
    dateFinished
  }
}
