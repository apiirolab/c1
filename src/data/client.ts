import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
