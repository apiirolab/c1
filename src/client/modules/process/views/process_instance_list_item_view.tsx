import * as React from 'react';

import { Link } from '@reach/router';
import { Button, List } from 'semantic-ui-react';

import { QueryTypes } from 'data/client';
import { ProcessIcon } from './process_icon';

type Props = {
  processInstance: QueryTypes.BpmnProcessInstancesBpmnProcessInstances;
};

export const ProcessInstanceListItem: React.SFC<Props> = ({ processInstance }) => (
  <List.Item>
    <List.Content floated="right">
      <Button primary icon="rocket" content="Launch" />
    </List.Content>
    <ProcessIcon type={processInstance.process.type} />
    <List.Content verticalAlign="top">
      <List.Header as={Link} to="nowhere">
        {processInstance.process.name}
      </List.Header>
      <List.Description>{processInstance.process.description}</List.Description>
    </List.Content>
  </List.Item>
);
