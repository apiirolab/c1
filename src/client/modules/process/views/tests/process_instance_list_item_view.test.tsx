import * as React from 'react';

import { QueryTypes, render } from 'client/tests';

import { ProcessInstanceListItem } from '../process_instance_list_item_view';

describe('Process', () => {
  describe('Instances', () => {
    describe('Item', () => {
      it('renders correctly', () => {
        const processInstance: QueryTypes.BpmnProcessInstancesBpmnProcessInstances = {
          id: '1',
          dateStarted: new Date(),
          process: {
            description: 'eqweqwe',
            name: 'eewrw',
            type: QueryTypes.ProcessType.Purchases
          },
          status: QueryTypes.BpmnProcessInstanceStatus.Finished
        };

        const component = render(<ProcessInstanceListItem processInstance={processInstance} />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
