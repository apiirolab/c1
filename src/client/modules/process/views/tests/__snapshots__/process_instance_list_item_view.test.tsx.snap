// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Instances Item renders correctly 1`] = `
<div>
  Empty1 12234234 weuweruwru
</div>
`;
